config/config.go

```diff
@@ -261,13 +283,35 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
 		return addrF(h.AllAddrs())
 	})}
 	if cfg.AutoNATService {
-		dialerStore := pstoremem.NewPeerstore()
-		dialerHost, err := cfg.newHost(ctx, dialerStore)
+		// Pull out the pieces of the config that we _actually_ care about.
+		// Specifically, don't setup things like autorelay, listeners,
+		// identify, etc.
+		autoNatCfg := Config{
+			Transports:         cfg.Transports,
+			Muxers:             cfg.Muxers,
+			SecurityTransports: cfg.SecurityTransports,
+			Insecure:           cfg.Insecure,
+			PSK:                cfg.PSK,
+			Filters:            cfg.Filters,
+			Reporter:           cfg.Reporter,
+
+			Peerstore: pstoremem.NewPeerstore(),
+		}
+		dialer, err := autoNatCfg.makeSwarm(ctx)
 		if err != nil {
 			h.Close()
 			return nil, err
 		}
-
+		dialerHost := blankhost.NewBlankHost(dialer)
+		err = autoNatCfg.addTransports(ctx, dialerHost)
```

Comment:
addTransports looks like it could plausibly be refactored to take a network directly, in which case we could just pass the dialer directly without ever creating the blank host. would that be more efficient?

Reply:
Unfortunately, _transports_ can currently accept a `Host` through reflection magic and we can't really switch that to `Network` because some transports might need to register a stream listener (only available on host).